internal/pkg/application/decoder/decoders.sensefarm.go

```diff
@@ -113,67 +115,81 @@ func SensefarmBasicDecoder(ctx context.Context, msg []byte, fn func(context.Cont
 }
 
 type payloadSensefarm struct {
-	TransmissionReason uint     // 	0 = unknown reset, 1 = POR/PDR reset, 2 = Independt watchdog reset, 3 = windows watchdog reset, 4 = low power reset, 5 = POR/PDR reset, 6 = Normal transmission, 7 = Button reset
-	ProtocolVersion    uint     // Version 0 -> 65535
-	BatteryVoltage     uint     // 0 -> 65535 mV
-	Resistance         []uint64 // 0 -> 4294967295 Ohm
-	SoilMoisture       []uint32 // 0 -> 65535 kPa. If value is too low or too high (sensor not placed outdoors, cable broke, etc), the sensor data is considered invalid and may not be sent./
-	Temperature        float32  // °C
+	TransmissionReason int8    // 	0 = unknown reset, 1 = POR/PDR reset, 2 = Independt watchdog reset, 3 = windows watchdog reset, 4 = low power reset, 5 = POR/PDR reset, 6 = Normal transmission, 7 = Button reset
+	ProtocolVersion    int16   // Version 0 -> 65535
+	BatteryVoltage     int16   // 0 -> 65535 mV
+	Resistances        []int32 // 0 -> 4294967295 Ohm
+	SoilMoistures      []int16 // 0 -> 65535 kPa. If value is too low or too high (sensor not placed outdoors, cable broke, etc), the sensor data is considered invalid and may not be sent./
+	Temperature        float32 // °C
 }
 
 func decodeSensefarmPayload(b []byte, p *payloadSensefarm) error {
 
-	readable_bytes := string(b)
-	readable_str := readable_bytes + " decoded as"
-	decode_state := "decode_header_byte"
-	println("Decoded string " + string(b))
+	for i := 0; i < len(b); i++ { //The multisensor message are read byte by byte and parsed for information on each individual sensor and it's values.
+		switch (b[i] & 0xFF) >> 3 {
+		case 0x01: //  Temperature
+			noOfBytes := 4
+			err := binary.Read(bytes.NewReader(b[i+1:]), binary.BigEndian, &p.Temperature)
+			if err != nil {
+				return err

```

Comment:
Här (och i efterföljande case) kan det vara en idé att dekorera det err som returneras med lite info om vad som försökte göras. Tex err = fmt.Errorf("failed to read temperature: %w", err)